### PR TITLE
Owncloud fix ldap config

### LINKED ISSUE #1741: [owncloud] error on configure ldap (nextcloud)
Setting up nextcloud with ldap integration fails on first run with:

```
TASK [owncloud : Check default ldap config exists] ***************************************************************************************************************************************************************
fatal: [nextcloud.intern.iris-ev.de]: FAILED! => {"changed": false, "cmd": ["php", "--file", "/srv/www/nextcloud//occ", "ldap:show-config", "s01"], "delta": "0:00:00.318695", "end": "2021-06-16 14:47:43.496665", "msg": "non-zero return code", "rc": 1, "start": "2021-06-16 14:47:43.177970", "stderr": "", "stderr_lines": [], "stdout": "Invalid configID", "stdout_lines": ["Invalid configID"]}
```

This is because the return code on an inexistent ldap config with gives an non null exit code:

```root@nextcloud:~# occ ldap:show-config  xx
Invalid configID
root@nextcloud:~# echo $?
1
```

and ansible fails.

No idea if this changed on recent nextcloud versions (I test with 21).

The solution would be e.g. to add `ignore_errors: True` to `tasks/ldap.yml`

```
- name: Check default ldap config exists
  command: php --file "{{ owncloud__app_home }}/occ" ldap:show-config "{{ owncloud__ldap_config_id }}"
  changed_when: False
  when: (not ansible_check_mode)
  register: owncloud__register_ldap_default_config_exits
  become: True
  become_user: '{{ owncloud__app_user }}'
  ignore_errors: True
```

### PR DESCRIPTION
**Pull request by jankowa**
should fix #1741

### CODE REVIEW

**Review by ypid — CHANGES_REQUESTED** [review]
Can you fix linting issue? Otherwise this works for me.

**Comment by jankowa** [comment]
> Can you fix linting issue? Otherwise this works for me.

yes. I'll do. I ignored them since they are not related to my commit.

**Comment by jankowa** [comment]
> > Can you fix linting issue? Otherwise this works for me.
> yes. I'll do. I ignored them since they are not related to my commit.
ah, you mean ansible-lint not yamllint. Yes there was still this trailing white space error. I thought I corrected it already.

File: ansible/roles/owncloud/tasks/ldap.yml [hunk 1]
```diff
@@ -14,6 +14,9 @@
   register: owncloud__register_ldap_default_config_exits
   become: True
   become_user: '{{ owncloud__app_user }}'
+  #ignore_errors: True
```
Comment (ypid) [inline ansible/roles/owncloud/tasks/ldap.yml]: What is the purpose of this commented out line?

Comment (jankowa) [reply ansible/roles/owncloud/tasks/ldap.yml]: just left over. fixed it.

File: ansible/roles/owncloud/tasks/ldap.yml [hunk 2]
```diff
@@ -14,6 +14,8 @@
   register: owncloud__register_ldap_default_config_exits
   become: True
   become_user: '{{ owncloud__app_user }}'
+  failed_when: (owncloud__register_ldap_default_config_exits.rc != 0 and
```
Comment (ypid) [inline ansible/roles/owncloud/tasks/ldap.yml:17]: Sorry for the ~3 time back and forth. `owncloud__register_ldap_default_config_exits` should probably mean `owncloud__register_ldap_default_config_exists`.

**Review by ypid — APPROVED** [review]
LGTM. I will fix the `owncloud__register_ldap_default_config_exits` -> `owncloud__register_ldap_default_config_exists` typo that I just noticed in a later patchset.

**Review by drybjed — APPROVED** [review]
Thanks for working on this issue an providing a good fix! I'll backport this to the stable release as well.

**Pull request merged**